### ISSUE TITLE
CombineToPDF does not work in search on Documentum and Hadoop

### ISSUE THREAD

**Issue by rosenoam** [tsgrp/HPI#1303]
When doing the combine to pdf action from search it thinks that all the word documents do not have a pdf rendition. This only occurs on dctm and hadoop
![combinepdf from search](https://cloud.githubusercontent.com/assets/22036850/22795948/27c2044a-eebe-11e6-8640-d168c9625881.gif)

**Comment by benallenallen** [comment]
Looks like some real shady code going on here that is looking for Alfresco specific renditions:

advancedcombinetopdf.js:
```
if(doc.get('mimeType') === "application/pdf" || (doc.get('properties').tsg_rendition && doc.get('properties').tsg_rendition === "application/pdf")){
```

In most cases, the native content will not be a PDF, so the reason it is working on Alfresco is because of the hardcoded checks for tsg:rendition="application/pdf". We really should be calling an endpoint to see if the documents each have a rendition rather than just relying on the oco to tell us if there is a pdf rendition (since Documentum and Hadoop impls don't in any way tell us if there is a PDF rendition). I understand the scaling issue and why it made sense for Alfresco to do it that way, but it is coded very repo specific right now, which is not a good thing.

**Comment by gsteimer** [comment]
So why does this work on Documentum from the stage?  It seems like it works fine there...

**Comment by b-k-white** [comment]
@gsteimer - In the stage it fetches the documents differently. It doesn't filter after it retrieves them:

                 ```
               //Grab all children documents of the current folder
				var container = new OC.OpenContentObject({
					'objectId': app.context.container.get("objectId")
				});
				//pass over getChildren's deferred
				deferred = container.getChildren(["application/pdf"]);
```

**Comment by gsteimer** [comment]
@b-k-white - is there a way we can do something similar when it's running from search?

**Comment by b-k-white** [comment]
We can't utilize the getChildren call with "application/pdf", because the documents have already been selected when it is a groupAction. We aren't getting the children of a given folder, as these documents could be in many different folders.

**Comment by rosenoam** [comment]
Fixed with HPI revision#17258   and OC revision#15982
